deoxys/src/modes.rs

```diff
@@ -294,38 +294,47 @@ where
         subkeys: &Array<DeoxysKey, B::SubkeysSize>,
         tag: &mut Tag,
     ) {
-        if !buffer.is_empty() {
-            tweak[0] = TWEAK_M;
+        if buffer.is_empty() {
+            return;
+        }
+        tweak[0] = TWEAK_M;
 
-            for (index, data) in buffer.chunks(16).enumerate() {
-                // Copy block number
-                tweak[8..].copy_from_slice(&(index as u64).to_be_bytes());
+        let (chunks, tail) = Block::slice_as_chunks(buffer);
+        let chunks_len = chunks.len();
 
-                if data.len() == 16 {
-                    let mut block = Block::default();
-                    block.copy_from_slice(data);
+        for (index, data) in chunks.into_iter().enumerate() {
```

Comment:
Hit approve then saw the clippy warning. It seems like if you use `chunks.iter()` rather than `chunks.into_iter()`, it won't consume `chunks`, and then you can just use `chunks.len()`